protocol/communities_messenger_test.go

```diff
@@ -2102,57 +2103,51 @@ func (s *MessengerCommunitiesSuite) TestLeaveAndRejoinCommunity() {
 
 func (s *MessengerCommunitiesSuite) TestShareCommunity() {
 	description := &requests.CreateCommunity{
-		Membership:  protobuf.CommunityPermissions_AUTO_ACCEPT,
+		Membership:  protobuf.CommunityPermissions_MANUAL_ACCEPT,
 		Name:        "status",
-		Color:       "#ffffff",
 		Description: "status community description",
+		Color:       "#FFFFFF",
+		Image:       "../_assets/tests/status.png",
+		ImageAx:     0,
+		ImageAy:     0,
+		ImageBx:     256,
+		ImageBy:     256,
+		Banner: images.CroppedImage{
+			ImagePath: "../_assets/tests/IMG_1205.HEIC.jpg",
+			X:         0,
+			Y:         0,
+			Width:     160,
+			Height:    90,
+		},
 	}
 
-	inviteMessage := "invite to community testing message"
+	response, err := s.owner.CreateCommunity(description, true)
 
-	// Create an community
-	response, err := s.bob.CreateCommunity(description, true)
 	s.Require().NoError(err)
 	s.Require().NotNil(response)
 	s.Require().Len(response.Communities(), 1)
-
+	s.Require().Len(response.Chats(), 1)
 	community := response.Communities()[0]
 
-	response, err = s.bob.ShareCommunity(
-		&requests.ShareCommunity{
-			CommunityID:   community.ID(),
-			Users:         []types.HexBytes{common.PubkeyToHexBytes(&s.alice.identity.PublicKey)},
-			InviteMessage: inviteMessage,
-		},
-	)
+	// Alice shares community with Bob
+	response, err = s.owner.ShareCommunity(&requests.ShareCommunity{
+		CommunityID:   community.ID(),
+		Users:         []types.HexBytes{common.PubkeyToHexBytes(&s.alice.identity.PublicKey)},
+		InviteMessage: "Come on alice, You'll like it here!",
+	})
+
 	s.Require().NoError(err)
 	s.Require().NotNil(response)
 	s.Require().Len(response.Messages(), 1)
+	sentMessageText := response.Messages()[0].Text
 
-	// Add bob to contacts so it does not go on activity center
-	bobPk := common.PubkeyToHex(&s.bob.identity.PublicKey)
-	request := &requests.AddContact{ID: bobPk}
-	_, err = s.alice.AddContact(context.Background(), request)
-	s.Require().NoError(err)
-
-	// Pull message and make sure org is received
-	err = tt.RetryWithBackOff(func() error {
-		response, err = s.alice.RetrieveAll()
-		if err != nil {
-			return err
-		}
-		if len(response.messages) == 0 {
-			return errors.New("community link not received")
-		}
-		return nil
-	})
-
+	_, err = WaitOnMessengerResponse(s.alice, func(r *MessengerResponse) bool {
+		return len(r.Messages()) > 0
+	}, "Messages not received")
 	s.Require().NoError(err)
 	s.Require().Len(response.Messages(), 1)
-
-	message := response.Messages()[0]
-	s.Require().Equal(community.IDString(), message.CommunityID)
-	s.Require().Equal(inviteMessage, message.Text)
+	s.Require().Contains(response.Messages()[0].Text, sentMessageText)
```

Comment:
Did you decide not to check `<message-text>\n<url>` structure?

Reply:
Sorry I forgot to add it into the test, Added.